### PR TITLE
[businessinsider] update regex (fixes #22929)

### LINKED ISSUE #22929: JW player downloads seem to be broken
Using the latest version 2019.10.29
Tried downloading JWPlayer video embedded on this page:
https://www.businessinsider.com/sesame-street-mock-trump-best-moments-budget-proposal-pbs-public-funding-2017-3

Here's the verbose output:

[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'-F', u'https://www.businessinsider.com/sesame-street-mock-trump-best-moments-budget-proposal-pbs-public-funding-2017-3?fbclid=IwAR2jOlaH5ADErmCQj44J8BOE-IJfNPAhBgnFPpV-nIOi7DK86sscO4YN9pA']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.10.29
[debug] Python version 2.7.16 (CPython) - Darwin-18.7.0-x86_64-i386-64bit
[debug] exe versions: rtmpdump 2.4
[debug] Proxy map: {}
[BusinessInsider] sesame-street-mock-trump-best-moments-budget-proposal-pbs-public-funding-2017-3: Downloading webpage
ERROR: Unable to extract jwplatform id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/businessinsider.py", line 39, in _real_extract
    webpage, 'jwplatform id')
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1005, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract jwplatform id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

### PR DESCRIPTION
**Pull request by uno20001**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR updates the regex used in the `businessinsider` extractor. Fixes #22929.

### CODE REVIEW

File: youtube_dl/extractor/businessinsider.py [hunk 1]
```diff
@@ -11,32 +11,43 @@ class BusinessInsiderIE(InfoExtractor):
...
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
-        jwplatform_id = self._search_regex(
-            (r'data-media-id=["\']([a-zA-Z0-9]{8})',
-             r'id=["\']jwplayer_([a-zA-Z0-9]{8})',
-             r'id["\']?\s*:\s*["\']?([a-zA-Z0-9]{8})'),
-            webpage, 'jwplatform id')
```
Comment (dstftw) [inline youtube_dl/extractor/businessinsider.py]: Do not touch the old patterns.

File: youtube_dl/extractor/businessinsider.py [hunk 2]
```diff
@@ -11,32 +11,43 @@ class BusinessInsiderIE(InfoExtractor):
...
+        
+        jwplatform_id = self._search_regex(r'(?:jwplatform\.com/players/|jwplayer_)([a-zA-Z0-9]+)', webpage, 'jwplatform id')
```
Comment (dstftw) [inline youtube_dl/extractor/businessinsider.py]: Read coding conventions.

**Comment by uno20001** [comment]
Hopefully, the code is now acceptable, I tried to comply with the coding conventions as much as possible, and tried to minimize the number of changes.

File: youtube_dl/extractor/businessinsider.py [hunk 3]
```diff
@@ -9,24 +9,37 @@ class BusinessInsiderIE(InfoExtractor):
...
-        'only_matching': True,
+        'md5': '4e81f3ae957d1e2e7e142f26a1bfaa95',
+        'info_dict': {
+            'id': '0TO9B5o6',
+            'ext': 'mp4',
+            'title': 'This is what separates the Excel masters from the wannabes',
+            'description': 'md5:af0ff0295b35d60bb336ba6eb5c79ecf',
+            'upload_date': '20171017',
+            'timestamp': 1508245260,
+        },
```
Comment (dstftw) [inline youtube_dl/extractor/businessinsider.py]: This test is identical to the first. Revert.

File: youtube_dl/extractor/businessinsider.py [hunk 4]
```diff
@@ -35,8 +48,10 @@ def _real_extract(self, url):
         jwplatform_id = self._search_regex(
             (r'data-media-id=["\']([a-zA-Z0-9]{8})',
              r'id=["\']jwplayer_([a-zA-Z0-9]{8})',
-             r'id["\']?\s*:\s*["\']?([a-zA-Z0-9]{8})'),
+             r'id["\']?\s*:\s*["\']?([a-zA-Z0-9]{8})',
+             r'(?:jwplatform\.com/players/|jwplayer_)([a-zA-Z0-9]+)'),
```
Comment (dstftw) [inline youtube_dl/extractor/businessinsider.py]: Video id length is 8.

File: youtube_dl/extractor/businessinsider.py [hunk 5]
```diff
@@ -35,8 +48,10 @@ def _real_extract(self, url):
...
             webpage, 'jwplatform id')
+
```
Comment (dstftw) [inline youtube_dl/extractor/businessinsider.py]: Remove.

**Comment by uno20001** [comment]
Thank you for your patience.

**Pull request merged**